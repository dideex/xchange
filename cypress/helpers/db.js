const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
