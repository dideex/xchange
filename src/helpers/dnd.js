import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
