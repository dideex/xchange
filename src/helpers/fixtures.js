export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}