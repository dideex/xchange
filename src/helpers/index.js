export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
