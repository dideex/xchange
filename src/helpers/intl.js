import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
