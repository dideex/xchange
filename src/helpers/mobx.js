import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
