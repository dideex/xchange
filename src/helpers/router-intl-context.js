import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
