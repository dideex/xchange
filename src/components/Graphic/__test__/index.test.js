import React from 'react'
import Component from '../index'
import {shallow} from 'enzyme'
import {mountWithIntl} from '../../../helpers/intl'

jest.mock('../../../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

describe('Graphic: index tests', () => {
  const fetch = () =>
    Promise.resolve({
      json() {
        return {
          values: [
            {x: new Date().setDate(1), y: 10},
            {x: new Date().setDate(2), y: 20},
            {x: new Date().setDate(3), y: 30},
            {x: new Date().setDate(4), y: 5},
          ],
        }
      },
    })
  window.fetch = fetch
  describe('Markup', () => {
    const wrapper = mountWithIntl(<Component />)
    it('Basic markup', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
