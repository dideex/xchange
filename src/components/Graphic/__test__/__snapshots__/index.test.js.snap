// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphic: index tests Markup Basic markup 1`] = `"<section class=\\"css-eshphd\\"><h2 class=\\"css-x184e\\"><span>Bitcoin rate for the last month</span></h2><div class=\\"css-xtttla\\"><div class=\\"recharts-responsive-container\\" style=\\"width: 100%; height: 450px;\\"><div style=\\"position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;\\"></div></div></div><svg class=\\"chart__svg\\" viewBox=\\"0 0 316 430\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M239.972 411.602L242.066 423.29C242.066 423.29 256.076 431.154 264.959 429.856C271.314 424.733 257.448 415.859 257.448 415.859L256.509 411.674\\" fill=\\"#35291F\\"></path><path d=\\"M0 415.786H7.0049L22.8923 246.31H14.0098L0 415.786Z\\" fill=\\"#A69281\\"></path><path d=\\"M114.462 415.786H122.405L135.476 246.31H126.377L114.462 415.786Z\\" fill=\\"#A69281\\"></path><path d=\\"M168.695 415.786H178.517L168.695 246.31H173.606L162.629 38.0894L143.925 246.31H158.874L168.695 415.786Z\\" fill=\\"#A69281\\"></path><path d=\\"M5.92163 246.31H143.925L162.629 38.0894H22.8923L5.92163 246.31Z\\" fill=\\"#BBCADB\\"></path><path d=\\"M220.979 60.5273C220.979 60.5273 223.001 55.6934 220.979 53.2403H214.624C210.436 42.1295 208.486 30.5136 212.097 17.4547C222.135 17.599 231.884 17.3104 241.344 17.4547C241.705 24.5974 244.016 28.5656 248.855 28.782C254.776 35.4918 245.749 38.1613 244.811 42.7067C244.305 47.9014 243.799 53.3125 243.222 59.1565L227.984 78.1315L220.979 60.5273Z\\" fill=\\"#E7D6BA\\"></path><path d=\\"M202.781 184.262H271.313C268.208 162.113 269.652 140.973 275.646 120.772C284.601 128.42 289.656 139.458 292.473 152.517L288.14 158.433L298.755 168.751C310.454 167.74 321.431 165.72 313.054 137.655C312.187 119.978 299.044 101.58 281.929 82.894C281.929 82.894 275.719 67.9593 247.049 63.6304C232.823 74.3805 223.579 75.1741 219.029 66.7327C204.659 68.32 195.343 84.1927 178.517 88.5216C173.245 88.4494 168.551 77.6272 164.001 63.6304L166.601 113.485C179.022 118.68 188.843 114.062 197.87 105.981C200.109 135.707 203.286 160.959 202.781 184.262Z\\" fill=\\"#BBCADB\\"></path><path d=\\"M288.14 158.36L269.58 188.879C279.257 185.632 289.006 178.923 298.683 168.678L288.14 158.36Z\\" fill=\\"#E7D6BA\\"></path><path d=\\"M129.41 38.0894C135.476 23.1547 151.364 28.2772 156.058 38.0894H129.41Z\\" fill=\\"#E7D6BA\\"></path><path d=\\"M218.957 66.5879L220.979 60.5274L228.49 72.2876L245.316 57.2808L247.049 63.4855L227.984 75.4622L218.957 66.5879Z\\" fill=\\"#394B5F\\"></path><path d=\\"M202.781 184.262C188.988 229.139 178.517 316.799 168.19 407.489H190.143L202.781 295.876C213.902 275.458 223.001 248.041 231.739 219.254L239.972 411.602H259.326L271.313 184.262H202.781Z\\" fill=\\"#394B5F\\"></path><path d=\\"M248.855 28.7101V7.78711C230.367 -0.654255 215.274 -3.46804 207.403 5.62265C207.403 11.1059 208.197 16.012 211.736 18.6815L241.344 17.455C241.85 25.247 244.594 28.638 248.855 28.7101Z\\" fill=\\"#394B5F\\"></path><path d=\\"M190.143 407.417L187.832 420.765H148.114C141.831 417.158 165.879 414.127 168.19 407.417H190.143Z\\" fill=\\"#35291F\\"></path><path d=\\"M34.158 177.769H22.0258L26.792 137.798H38.9242L34.158 177.769Z\\" fill=\\"white\\"></path><path d=\\"M47.59 177.769H35.4578L38.563 151.868H50.6952L47.59 177.769Z\\" fill=\\"#B15654\\"></path><path d=\\"M77.4872 177.768H65.355L72.6487 119.112H84.7809L77.4872 177.768Z\\" fill=\\"#B15654\\"></path><path d=\\"M111.14 177.769H99.0073L108.034 106.63H120.166L111.14 177.769Z\\" fill=\\"#B15654\\"></path><path d=\\"M64.3439 177.769H52.1395L58.2056 125.894H70.3378L64.3439 177.769Z\\" fill=\\"white\\"></path><path d=\\"M97.6354 177.769H85.5032L88.5362 151.868H100.741L97.6354 177.769Z\\" fill=\\"white\\"></path><path d=\\"M136.415 177.768H124.211L132.082 112.474H144.214L136.415 177.768Z\\" fill=\\"white\\"></path></svg></section>"`;
